config-service-impl/src/main/java/org/hypertrace/config/service/store/DocumentConfigStore.java

```diff
@@ -76,6 +79,61 @@ public UpsertedConfig writeConfig(
     Optional<ConfigDocument> previousConfigDoc = getLatestVersionConfigDoc(configResourceContext);
     Optional<ContextSpecificConfig> optionalPreviousConfig =
         previousConfigDoc.flatMap(this::convertToContextSpecificConfig);
+
+    Key latestDocKey = new ConfigDocumentKey(configResourceContext);
+    ConfigDocument latestConfigDocument =
+        buildConfigDocument(configResourceContext, latestConfig, userId, previousConfigDoc);
+
+    collection.upsert(latestDocKey, latestConfigDocument);
+    return optionalPreviousConfig
+        .map(previousConfig -> this.buildUpsertResult(latestConfigDocument, previousConfig))
+        .orElseGet(() -> this.buildUpsertResult(latestConfigDocument));
+  }
+
+  private List<UpsertedConfig> writeConfigs(
+      Map<ConfigResourceContext, Value> resourceContextValueMap, String userId) throws IOException {
+    Map<ConfigResourceContext, Optional<ConfigDocument>> previousConfigDocs =
+        getLatestVersionConfigDocs(resourceContextValueMap.keySet());
+    Map<Key, Document> documentsToBeUpserted = new LinkedHashMap<>();
+    for (Entry<ConfigResourceContext, Optional<ConfigDocument>> configResourceContextOptionalEntry :
+        previousConfigDocs.entrySet()) {
+      ConfigResourceContext configResourceContext = configResourceContextOptionalEntry.getKey();
```

Comment:
nit: lots of nice iteration options on a map - like a foreach gives you a function accepting a key and value I believe.